### PR TITLE
Virtual Set 16

### PR DESCRIPTION
**Pull request by Gergall**
VSet 16 plus a few small fixes.

### CODE REVIEW

**Comment by ronhowe** [comment]
Could you add Set 16 to the sets.json file too?

**Comment by ronhowe** [comment]
Escape Pod & We're Doomed 

The JSON references a card in a hires subfolder.

card "/starwars/Virtual4-Light/hires/escapepodweredoomed.png"

It is technically there, but the only card I can find that does this.

Is this a new option going forward that I need to code for?  (I use image pathing to do psuedo-sorting and this throws it off.)

**Comment by ronhowe** [comment]
> Escape Pod & We're Doomed
> 
> The JSON references a card in a hires subfolder.
> 
> card "/starwars/Virtual4-Light/hires/escapepodweredoomed.png"
> 
> It is technically there, but the only card I can find that does this.
> 
> Is this a new option going forward that I need to code for? (I use image pathing to do psuedo-sorting and this throws it off.)

There are the standard images in the root (thumbnail) and large subfolders.

So I'm not sure why this is necessary unless there is a grand scheme in play.

**Comment by ronhowe** [comment]
> > Escape Pod & We're Doomed
> > The JSON references a card in a hires subfolder.
> > card "/starwars/Virtual4-Light/hires/escapepodweredoomed.png"
> > It is technically there, but the only card I can find that does this.
> > Is this a new option going forward that I need to code for? (I use image pathing to do psuedo-sorting and this throws it off.)
> 
> There are the standard images in the root (thumbnail) and large subfolders.
> 
> So I'm not sure why this is necessary unless there is a grand scheme in play.

Correct image reference:

https://res.starwarsccg.org/cards/Virtual4-Light/large/escapepodweredoomed.gif

File: Light.json [hunk 1]
```diff
@@ -57774,6 +57774,1068 @@
...
+          "set": "216"
+        }
+      ],
+      "front": {
+        "title": "•Jedi Holocron",
+        "imageUrl": "https://res.starwarsccg.org/cards/Virtual16-Light/large/jediholocron.gif",
+        "type": "Device",
+        "uniqueness": "*",
+        "destiny": "2",
+        "gametext": "Deploy on your character of ability > 4. While present: adds 1 to training destiny draws and Force drains here; the first Force lost to a Force drain here is stacked here face down; opponent’s ability required to draw battle destiny here is +1 for each card stacked here.",
```
Comment (ronhowe) [inline Light.json:58197]: Please use a single ' mark instead of the curly tick mark.

File: Light.json [hunk 2]
```diff
@@ -16003,11 +16003,11 @@
       ],
       "front": {
         "title": "Escape Pod & We're Doomed",
-        "imageUrl": "https://res.starwarsccg.org/cards/Virtual4-Light/large/escapepodweredoomed.gif",
+        "imageUrl": "https://res.starwarsccg.org/cards/Virtual4-Light/hires/escapepodweredoomed.png",
```
Comment (ronhowe) [inline Light.json:16006]: Proposed standard image URI:

https://res.starwarsccg.org/cards/Virtual4-Light/large/escapepodweredoomed.gif

**Comment by DevoKun** [comment]
Hey, Ron.
If you look at the [**Card Image docs**](https://github.com/swccgpc/holotable/blob/master/docs/card-images.md) in the holotable repo you will see that the `hires` images are the source of truth. Everything flows downhill from there.



There is a desire to use the `hires` images with "*gemp**.

That being said, you are right that the json db should continue to reference the **large** images. Alternatively we need to update the downstream parsers to replace `hires` with `large`. What do you think?

**Comment by ronhowe** [comment]
> Hey, Ron.
> If you look at the [**Card Image docs**](https://github.com/swccgpc/holotable/blob/master/docs/card-images.md) in the holotable repo you will see that the `hires` images are the source of truth. Everything flows downhill from there.
> 
> 
> 
> There is a desire to use the `hires` images with **gemp**. I could see it both ways, but might be valuable to have the JSON DB reference the `hires` image. The `hires` images are not always available, so this would be any easy way of knowing without checking. The downside is that oversized images would be used by systems blind referencing the json db.
> 
> That being said, you are right that the json db should continue to reference the **large** images. Alternatively we need to update the downstream parsers to replace `hires` with `large`. What do you think?

I see.

In this instance, I think everything is in order *except* this one file referencing the hires image.  The large image does exist and should be referenced like all the others.

**Comment by ronhowe** [comment]
> Hey, Ron.
> If you look at the [**Card Image docs**](https://github.com/swccgpc/holotable/blob/master/docs/card-images.md) in the holotable repo you will see that the `hires` images are the source of truth. Everything flows downhill from there.
> 
> 
> 
> There is a desire to use the `hires` images with **gemp**. I could see it both ways, but might be valuable to have the JSON DB reference the `hires` image. The `hires` images are not always available, so this would be any easy way of knowing without checking. The downside is that oversized images would be used by systems blind referencing the json db.
> 
> That being said, you are right that the json db should continue to reference the **large** images. Alternatively we need to update the downstream parsers to replace `hires` with `large`. What do you think?

Also, something didn't work with generating the thumbnails on the light side.  There are only 2.  The rest are missing.

File: Light.json (hunk 1 above)
Comment (DevoKun) [reply Light.json:58197]: I replaced the smartquote in Light.json.
I also added a workflow to automatically fix smartquotes going forward.

**Comment by DevoKun** [comment]
I added the missing set16 to `sets.json`.

**Pull request merged**